akka-actor-typed/src/main/scala/akka/actor/typed/internal/adapter/ActorAdapter.scala

```diff
@@ -200,11 +190,9 @@ import akka.util.OptionVal
       case null                   ⇒ // skip PostStop
       case _: DeferredBehavior[_] ⇒
       // Do not undefer a DeferredBehavior as that may cause creation side-effects, which we do not want on termination.
-      case s: StoppedBehavior[_] ⇒ s.postStop match {
-        case OptionVal.Some(postStop) ⇒ Behavior.interpretSignal(postStop, ctx, PostStop)
-        case OptionVal.None           ⇒ // no postStop behavior defined
-      }
-      case b ⇒ Behavior.interpretSignal(b, ctx, PostStop)
+      case b ⇒
+        if (_ctx eq null) initializeContext()
```

Comment:
why is this needed? should it always be done in `preStart` instead?

Reply:
Yeah, makes sense to me, seems like we did some premature optimization on avoiding to initialize ctx in case of stopping immediately but that seems like a weird case to optimize to me.